### PR TITLE
Custom Connectors + HDT

### PR DESCRIPTION
**Pull request by bsteenwi**
request to make it possible to write own connector classes
+ example of how to define a HDT connector

The current HDT connector example implementation does not support all features (e.g. literal support)

### CODE REVIEW

**Comment by GillesVandewiele** [comment]
This looks great Bram! I'll do a more in-depth review later and will check for the failing CI checks (something wrong with the linting atm). Perhaps we could move the HDTConnector to its own file such that it is easier for others to use (and find)?

**Comment by GillesVandewiele** [comment]
This looks great Bram! I'll do a more in-depth review later and will check for the failing CI checks (something wrong with the linting atm). Perhaps we could move the HDTConnector to its own file such that it is easier for others to use (and find)?

**Comment by rememberYou** [comment]
Thanks Bram! Don't worry too much about the linter, it's because `black` doesn't work with a new version of `click`(cf., [issue](https://github.com/psf/black/issues/2964)). However, the dependencies have been updated on the develop branch and this problem no longer persists.

File: examples/mutag_hdt.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+entities = train_entities + test_entities
+labels = train_labels + test_labels
+
+
+@attr.s
+class HDTConnector(Connector):
+    def __attrs_post_init__(self):
+        self.store = None
+
+    @cachedmethod(operator.attrgetter("cache"), key=partial(hashkey, "fetch"))
```
Comment (rememberYou) [inline examples/mutag_hdt.py:29]: @bsteenwi Could you remove `key=partial(hashkey, "fetch")`? With the new version of `cachetool`, we have an issue by specifying the key.

`@cachedmethod(operator.attrgetter("cache"))` should be fine.

**Comment by bsteenwi** [comment]
> This looks great Bram! I'll do a more in-depth review later and will check for the failing CI checks (something wrong with the linting atm). Perhaps we could move the HDTConnector to its own file such that it is easier for others to use (and find)?

@GillesVandewiele We could create a separate file for the HDTConnector, but would it be better to create separate files for all our connectors?

**Comment by GillesVandewiele** [comment]
Yes, or bundle all of them in a connectors.py file!

**Pull request merged**